### PR TITLE
Add an Apple privacy info file for OpenSSL.

### PR DESCRIPTION
**Pull request by slontis**
##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Comment by slontis** [comment]
Fixes #23494

**Comment by t8m** [comment]
Should we treat it as a bug fix and add it on all branches?

**Comment by t8m** [comment]
Or documentation actually.

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
As @fwh-dc is the community maintainer of iOS platforms I kindly ask him to review this.

**Comment by fwh-dc** [comment]
LGTM.

I have one minor thing to consider:
Should we copy this file to the install location of ios build targets to make it more convenient when packaging an xcframework?

**Comment by kroeckx** [comment]
There is also a discussion about this at #23262

**Review by paulidale — APPROVED** [review]
I'm fine back merging this anywhere.

**Comment by slontis** [comment]
someone with better knowledge of the config/perl would need to add it to the install process.. At least this gives a reference file.

**Comment by t8m** [comment]
@fwh-dc Which directory it should be installed to?  The same where the libcrypto/libssl .dylib is installed to? BTW the iOS targets by default disable `shared` build so there is actually no .dylib built. So I would skip installing this anywhere for now. 

The iOS targets really need some fine-tuning/cleanups IMO to be more comprehensive and useful.

**Comment by fwh-dc** [comment]
> @fwh-dc Which directory it should be installed to? The same where the libcrypto/libssl .dylib is installed to? BTW the iOS targets by default disable `shared` build so there is actually no .dylib built. So I would skip installing this anywhere for now.

Good question. I think it's fine to just have the file for reference as suggested. We can always improve from there.

**Comment by MichaelWellsSM** [comment]
Doesn't this need to be added to the appropriate directory in the podspec and packaging (resource bundle) ?

**Comment by davidben** [comment]
Based on https://github.com/openssl/openssl/issues/23494#issuecomment-2036134844, I take it this was derived from the BoringSSL one? Keep in mind that `stat` appears in this list:
https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_use_of_required_reason_api?language=objc

We have an empty one, but we'd also removed our remaining `stat` call. I haven't looked too carefully at this, but it's possible you all may need either code changes or a different manifest.

**Comment by paulidale** [comment]
I do not see a need for us to install this.  It's useful documentation for folks building for Apple products.
This is very unlikely to stand by itself as the only things an application needs.

**Comment by t8m** [comment]
> Based on [#23494 (comment)](https://github.com/openssl/openssl/issues/23494#issuecomment-2036134844), I take it this was derived from the BoringSSL one? Keep in mind that `stat` appears in this list: https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_use_of_required_reason_api?language=objc
> 
> We have an empty one, but we'd also removed our remaining `stat` call. I haven't looked too carefully at this, but it's possible you all may need either code changes or a different manifest.

Yeah maybe we need to declare C617.1 Although we never use the timestamp info obtained with stat.

**Comment by t8m** [comment]
**OTC: We are OK with adding the file. The contents need to be right. It seems we need to declare C617.**

**Comment by kroeckx** [comment]
C617 is probably not right, and we might need to ask apple for a new id.

**Comment by kroeckx** [comment]
I guess I was confused, it looks right

**Comment by kroeckx** [comment]
Apple's description is unclear to me, but it seems like the best match given the documentation. For instance, how do we check certificates on iOS, and do those files belong to the app as the documentation seems to imply? It's also not clear if that code means it can be transferred off the device or not, which we don't do.

**Review by levitte — APPROVED** [review]

File: PrivacyInfo.xcprivacy [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<!--
+This is an Apple related Privacy Manifest File for OpenSSL as required by
+ https://developer.apple.com/support/third-party-SDK-requirements/
```
Comment (tom-cosgrove-arm) [inline PrivacyInfo.xcprivacy:6]: (minor) should there be a space at the beginning of this line?

**Comment by slontis** [comment]
Is there someone who actually has Xcode installed who can generate this file
I dont feel comfortable trying to add this entry manually - and I dont have access to Xcode + MAC to do this..

(Submit a new PR if you want)
Basically the NSPrivacyAccessedAPITypes section needs 
NSPrivacyAccessedAPIType=APICategoryFileTimeStamp
NSPrivacyAccessedAPITypeReasons=C617.1

**Comment by mattcaswell** [comment]
In light of #24260 can this be closed now?

**Comment by slontis** [comment]
I would think so.

**Pull request closed without merging**